realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -31,49 +30,80 @@
+        public long columnStringIndex;
+        public long columnLongIndex;
+        public long columnFloatIndex;
+        public long columnDoubleIndex;
+        public long columnBooleanIndex;
+        public long columnDateIndex;
+        public long columnBinaryIndex;
+        public long columnObjectIndex;
+        public long columnRealmListIndex;
 
         AllTypesColumnInfo(String path, Table table) {
             final Map<String, Long> indicesMap = new HashMap<String, Long>(9);
             this.columnStringIndex = getValidColumnIndex(path, table, "AllTypes", "columnString");
             indicesMap.put("columnString", this.columnStringIndex);
-
             this.columnLongIndex = getValidColumnIndex(path, table, "AllTypes", "columnLong");
             indicesMap.put("columnLong", this.columnLongIndex);
-
             this.columnFloatIndex = getValidColumnIndex(path, table, "AllTypes", "columnFloat");
             indicesMap.put("columnFloat", this.columnFloatIndex);
-
             this.columnDoubleIndex = getValidColumnIndex(path, table, "AllTypes", "columnDouble");
             indicesMap.put("columnDouble", this.columnDoubleIndex);
-
             this.columnBooleanIndex = getValidColumnIndex(path, table, "AllTypes", "columnBoolean");
             indicesMap.put("columnBoolean", this.columnBooleanIndex);
-
             this.columnDateIndex = getValidColumnIndex(path, table, "AllTypes", "columnDate");
             indicesMap.put("columnDate", this.columnDateIndex);
-
             this.columnBinaryIndex = getValidColumnIndex(path, table, "AllTypes", "columnBinary");
             indicesMap.put("columnBinary", this.columnBinaryIndex);
-
             this.columnObjectIndex = getValidColumnIndex(path, table, "AllTypes", "columnObject");
             indicesMap.put("columnObject", this.columnObjectIndex);
-
             this.columnRealmListIndex = getValidColumnIndex(path, table, "AllTypes", "columnRealmList");
             indicesMap.put("columnRealmList", this.columnRealmListIndex);
 
-            setIndicesMap(indicesMap);
+            setIndicesMap(Collections.unmodifiableMap(indicesMap));
         }
-    }
 
+        private AllTypesColumnInfo(AllTypesColumnInfo original) {
+            this.columnStringIndex = original.columnStringIndex;
+            this.columnLongIndex = original.columnLongIndex;
+            this.columnFloatIndex = original.columnFloatIndex;
+            this.columnDoubleIndex = original.columnDoubleIndex;
+            this.columnBooleanIndex = original.columnBooleanIndex;
+            this.columnDateIndex = original.columnDateIndex;
+            this.columnBinaryIndex = original.columnBinaryIndex;
+            this.columnObjectIndex = original.columnObjectIndex;
+            this.columnRealmListIndex = original.columnRealmListIndex;
+
+            setIndicesMap(original.getIndicesMap());
+        }
+
+        @Override
+        public final void copyFrom(ColumnInfo other) {
```

Comment:
I would probably rename this to `copyColumnInfoFrom(other)`. `copyFrom` makes me think it will copy the entire RealmObject.